ghcide/test/exe/Main.hs

```diff
@@ -175,14 +170,14 @@ instance Pretty Log where
 -- | Wait for the next progress begin step
 waitForProgressBegin :: Session ()
 waitForProgressBegin = skipManyTill anyMessage $ satisfyMaybe $ \case
-  FromServerMess SProgress (NotificationMessage _ _ (ProgressParams _ (Begin _))) -> Just ()
+  FromServerMess  SMethod_Progress  (TNotificationMessage _ _ (ProgressParams _ v)) | not $ Lens.isn't _workDoneProgressBegin v-> Just ()
```

Comment:
still vaguely sus about this. Or maybe we need to export a more useful helper from lsp?